mods/mc_teacher/gui_dash.lua

```diff
@@ -659,414 +657,348 @@ end
-				end
-			else
-				minetest.chat_send_player(pname,pname..": Please click on a classroom in the list to join.")
-			end
-		elseif fields.back then
-			show_teacher_menu(player)
-		elseif fields.delete then
-		-- TO DO: currently, this function only deletes the symbolic link to the classroom in the palyer/mod metadata. need to also remove the physical world.
-			-- Update player metadata
-			if context.selected then
-				pmeta = player:get_meta()
-				pdata = minetest.deserialize(pmeta:get_string("classrooms"))
-				-- Update modstorage first
-				mdata = minetest.deserialize(minetest_classroom.classrooms:get_string("classrooms"))
-				loc = check_access_code(pdata.access_code[context.selected],mdata.access_code)
-				mdata.course_code[loc] = nil
-				mdata.section_number[loc] = nil
-				mdata.start_year[loc] = nil
-				mdata.start_month[loc] = nil
-				mdata.start_day[loc] = nil
-				mdata.end_year[loc] = nil
-				mdata.end_month[loc] = nil
-				mdata.end_day[loc] = nil
-				mdata.classroom_map[loc] = nil
-				mdata.access_code[loc] = nil
-				mdata.spawn_pos[loc] = nil
-				minetest_classroom.classrooms:set_string("classrooms",minetest.serialize(mdata
-				))
-				
-				-- Update player metadata
-				pdata.course_code[context.selected] = nil
-				pdata.section_number[context.selected] = nil
-				pdata.start_year[context.selected] = nil
-				pdata.start_month[context.selected] = nil
-				pdata.start_day[context.selected] = nil
-				pdata.end_year[context.selected] = nil
-				pdata.end_month[context.selected] = nil
-				pdata.end_day[context.selected] = nil
-				pdata.classroom_map[context.selected] = nil
-				pdata.access_code[context.selected] = nil
-				pdata.spawn_pos[context.selected] = nil
-				pmeta:set_string("classrooms",minetest.serialize(pdata
-				))
-				show_classrooms(player)
-			else
-				minetest.chat_send_player(pname,pname..": Please click on a classroom in the list to delete.")
-			end
-		-- TODO: eventually delete this button, only useful for testing
-		elseif fields.deleteall then
-			pmeta:set_string("classrooms", nil)
-			minetest_classroom.classrooms:set_string("classrooms", nil)
-			show_classrooms(player)
-		else -- escape without input
-			return
-		end
-	end
+    local wait = os.clock()
+    while os.clock() - wait < 0.05 do
+    end --popups don't work without this
+
+    local pname = player:get_player_name()
+
+    -- Menu
+    if formname == "mc_teacher:menu" then
+        if fields.spawn then
+
+            local spawnRealm = mc_worldManager.GetSpawnRealm()
+            local spawn_pos = spawnRealm.SpawnPoint
+            player:set_pos(spawn_pos)
+        elseif fields.tasks then
+            show_tasks(player)
+        elseif fields.lessons then
+            show_lessons(player)
+        elseif fields.classrooms then
+            show_classrooms(player)
+        elseif fields.players then
+            show_players(player)
+        elseif fields.mail then
+            show_mail(player)
+        end
```

Comment:
The code for handling the `fields.species` conditional (from `magnify`) seems to have been lost while resolving merge conflicts
```suggestion
        elseif fields.species then
            show_species(player)
        end
```